### PR TITLE
Allow spark to be installed as an s2i build

### PR DESCRIPTION
**Pull request by tmckayus**
This is an initial change to allow spark to be installed during
an s2i build if it is left out of the creation of the s2i images

This is a step toward a mechanism which will allow users to build s2i
images with custom spark installs in OpenShift Origin rather
than modifying github repos and running local builds.

### CODE REVIEW

File: java-build/modules/common/added/s2i/assemble [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+source $STI_SCRIPTS_PATH/s2i-env-vars
+set -x
+
+function match_sum {
+    local sumfile=$1
+    local delim=$2
+    local md5=$3
+    echo delim is "$delim"
```
Comment (tmckayus) [inline java-build/modules/common/added/s2i/assemble]: debug code, this should come out

File: java-build/modules/common/added/s2i/assemble [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    return 1 
+}
+
+if [ -f $SPARK_HOME/bin/spark-submit ]; then
+    echo "Spark is installed"
+else
+    echo "Attempting to install Spark"
+    
+    # If a url has been specfified for spark use it
+    env | grep SPARK
```
Comment (tmckayus) [inline java-build/modules/common/added/s2i/assemble]: debug code, this can come out

File: java-build/modules/common/added/s2i/assemble [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+        # Does the tarball contain a spark-submit?
+        name=$(tar -tzf $spark | grep "spark-submit$")
+        if [ "$?" -eq 0 ]; then
+
+	    # See if we have an md5 file to match against
+	    if [ -f "$spark".md5 ]; then
+		calcvalue=$(md5sum "$spark" | cut -d\  -f1)
+		# split the md5 file using a space
+                match_sum "$spark".md5 \  $calcvalue
```
Comment (tmckayus) [inline java-build/modules/common/added/s2i/assemble]: spacing is off

**Comment by crobby** [comment]
Seems to be working as advertised so far.  Amazing stuff!

**Pull request merged**